### PR TITLE
MRG: fix benchmark code & a few other small issues from pyOpenSci review

### LINKED ISSUE #2906: fix unused imports in benchmarks/benchmarks.py
per [review](https://github.com/pyOpenSci/software-submission/issues/129#issuecomment-1870572982)

neither `os` nor `Path` are used, despite being imported.

### LINKED ISSUE #2913: update benchmarks.py "magic numbers"
per [@LilyAnderssonLee! review](https://github.com/pyOpenSci/software-submission/issues/129#issuecomment-1838233643)

>Magic numbers: There are some magic numbers in the code (e.g., 500, 21, 3000, 10000, 1000, 20). Is it possible to define these numbers as constants with meaningful names or add the comments to make the code more self-explanatory?

### LINKED ISSUE #2914: fix download link in tutorial
per [@LilyAnderssonLee! review](https://github.com/pyOpenSci/software-submission/issues/129#issuecomment-1838233643)

>he download link doesn't work in the section [#Building your own LCA database](https://sourmash.readthedocs.io/en/latest/tutorials-lca.html). The correct link should be `curl -L https://osf.io/bw8d7/download -o delmont-subsample-sigs.tar.gz`

### LINKED ISSUE #2919: add explanations & default value to `sourmash lca classify -h`
per [@LilyAnderssonLee review](https://github.com/pyOpenSci/software-submission/issues/129#issuecomment-1838233643)

>Is it possible to add explanations and default values to `--threshold` in the help message of `sourmash lca classify -h`

### PR DESCRIPTION
**Pull request by ctb**
Address issues:
* https://github.com/sourmash-bio/sourmash/issues/2919
* https://github.com/sourmash-bio/sourmash/issues/2913
* https://github.com/sourmash-bio/sourmash/issues/2906
* https://github.com/sourmash-bio/sourmash/issues/2914

Fixes https://github.com/sourmash-bio/sourmash/issues/2919
Fixes https://github.com/sourmash-bio/sourmash/issues/2913
Fixes https://github.com/sourmash-bio/sourmash/issues/2906
Fixes https://github.com/sourmash-bio/sourmash/issues/2914

Remaining TODO:
* test `benchmarks/benchmarks.py`

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge @sourmash-bio/devs !

File: benchmarks/benchmarks.py [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
 from sourmash.minhash import MinHash
 
+RANDOM_SEQ_SIZE=3000
+RANDOM_SEQ_NUMBER=300
+
+MINHASH_NUM=500
+MINHASH_K=21
+
+GET_MINS_RANGE=500
+ADD_HASH_RANGE=10_000
```
Comment (ccbaumler) [inline benchmarks/benchmarks.py:14]: Will the underscore throw an error in `range()` ?

**Review by ccbaumler — COMMENTED** [review]
LGTM

File: benchmarks/benchmarks.py (hunk 1 above)
Comment (ctb) [reply benchmarks/benchmarks.py:14]: as it turns out, that is a valid Python way to write 10e5!

**Pull request merged**